wallet-core/src/transaction.rs

```diff
@@ -0,0 +1,336 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+//! Implementations of basic wallet functionalities to create transactions.
+
+use alloc::vec::Vec;
+
+use rand::{CryptoRng, RngCore};
+
+use ff::Field;
+use poseidon_merkle::Opening;
+use zeroize::Zeroize;
+
+use execution_core::{
+    signatures::bls::SecretKey as BlsSecretKey,
+    stake::{Stake, Withdraw as StakeWithdraw, STAKE_CONTRACT},
+    transfer::{
+        contract_exec::{ContractCall, ContractExec},
+        phoenix::{
+            Note, Prove, PublicKey as PhoenixPublicKey,
+            SecretKey as PhoenixSecretKey, Transaction as PhoenixTransaction,
+            NOTES_TREE_DEPTH,
+        },
+        withdraw::{Withdraw, WithdrawReceiver, WithdrawReplayToken},
+        Transaction,
+    },
+    BlsScalar, ContractId, JubJubScalar,
+};
+
+/// Types that are client of the prover.
+pub trait ProverClient {
```

Comment:
The ProverClient trait would be nice to have but will be problematic for the FFI unless we do wasm imports. A function to replace this would be nice and all the other transaction functions return an unproven transaction